### PR TITLE
[WIP]: Decouple InsecureServingInfo from genericConfig initialization

### PR DESCRIPTION
**Pull request by yue9944882**
**What this PR does / why we need it**:

Shorten`BuildGenenricConfig`'s long return list. `InsecureServingInfo` is initialized at the deepest layer but finally used on the top: [https://github.com/kubernetes/kubernetes/blob/007bf90e32733286a5eb293b579db880a702eb5b/cmd/kube-apiserver/app/server.go#L201](https://github.com/kubernetes/kubernetes/blob/007bf90e32733286a5eb293b579db880a702eb5b/cmd/kube-apiserver/app/server.go#L201)

xref #66386

/assign @deads2k 
/sig api-machinery

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: cmd/kube-apiserver/app/server.go [hunk 1]
```diff
@@ -197,7 +197,10 @@ func CreateServerChain(completedOptions completedServerRunOptions, stopCh <-chan
 		return nil, err
 	}
 
-	if insecureServingOptions != nil {
+	if true || completedOptions.InsecureServing.BindPort > 0 {
```
Comment (CaoShuFeng) [inline cmd/kube-apiserver/app/server.go]: What's the meaning of `if true || xxx` ?
Maybe add some comment here.

Comment (yue9944882) [reply cmd/kube-apiserver/app/server.go]: hi @CaoShuFeng thanks for reviewing. i thought that it's impossible to make the server run without insecure related options. reverted the change for better human-read

File: cmd/kube-apiserver/app/server.go [hunk 2]
```diff
@@ -197,7 +197,10 @@ func CreateServerChain(completedOptions completedServerRunOptions, stopCh <-chan
 		return nil, err
 	}
 
-	if insecureServingOptions != nil {
+	if completedOptions.InsecureServing != nil && completedOptions.InsecureServing.BindPort > 0 {
+		insecureServingOptions := &kubeserver.InsecureServingInfo{
```
Comment (sttts) [inline cmd/kube-apiserver/app/server.go]: can we rename to `insecureServingInfo`?

**Pull request closed without merging**